### PR TITLE
fix: handle decoding errors

### PR DESCRIPTION
**Pull request by 0xyaco**
# 🤖 Linear

Closes GRT-242

## Description

* Catches all errors thrown by decoding `bytes` properties read from Prophet events
* Removes `ProphetCodec.decode...` calls from ProtocolProvider `getXXXEvent` to use them during actual event processing and not event fetching

### CODE REVIEW

File: packages/automated-dispute/tests/services/eboMemoryRegistry/commands/addRequest.spec.ts [hunk 1]
```diff
@@ -78,6 +74,72 @@ describe("AddRequest", () => {
...
+                        ...request["prophetData"],
+                        requestModuleData: "0xbad" as Hex,
+                    },
+                },
+            };
+
+            try {
+                AddRequest.buildFromEvent(malformedRequestEvent, registry);
+
+                fail("Expecting error");
```
Comment (0xnigir1) [inline packages/automated-dispute/tests/services/eboMemoryRegistry/commands/addRequest.spec.ts:93]: ooo didn't know this trick

**Review by 0xnigir1 — COMMENTED** [review]
looking good 👍 , just some comments

File: packages/automated-dispute/src/exceptions/prophetDecodingError.exception.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ByteArray, Hex } from "viem";
+
+export class ProphetDecodingError extends Error {
+    constructor(
+        public readonly id: string,
+        public readonly data: ByteArray | Hex,
+        public readonly err?: Error,
```
[the remark this answers is not in the document]
Comment (0xyaco) [reply packages/automated-dispute/src/exceptions/prophetDecodingError.exception.ts:7]: Nice catch! The idea was to log the name of the original error to provide more context during debugging to the user. Ended up logging this `err`'s name too

File: packages/automated-dispute/src/services/prophetCodec.ts [hunk 3]
```diff
@@ -201,9 +237,31 @@ export class ProphetCodec {
...
+            if (!disputeStatus) throw new ProphetDecodingError("dispute.status", toHex(status));
+            else return disputeStatus;
+        } catch (err) {
+            throw new ProphetDecodingError(
+                "dispute.status",
+                toHex(status.toString()),
+                err instanceof Error ? err : undefined,
+            );
+        }
+    }
```
[the remark this answers is not in the document]
Comment (0xyaco) [reply packages/automated-dispute/src/services/prophetCodec.ts]: Big :brain:, fixed

**Review by 0xnigir1 — COMMENTED** [review]
just a commit suggestion and is good to go

**Review by 0xnigir1 — APPROVED** [review]

**Pull request merged**